### DISCUSSION TITLE
Get the output of the TIN algorithm

### DISCUSSION THREAD

**Discussion by jp136769** [r-lidar/lidR#772] [Q&A] [answered]
Hi,

I would like to get access to the TIN's nodes generated by the `tin()` algorithm for spatial interpolation.

Would it be possible to output a `data.table` with the unique XYZ coordinates and triangle ID for example?

What I need in definitive is the coordinates of the points that are used and interpolated to produce the raster with `rasterize_terrain()`.

Regards

**Comment by Jean-Romain** [comment, marked as answer]
Not, not easily. You could look at the source code and use internal function https://github.com/r-lidar/lidR/blob/master/R/utils_delaunay.R.

You can also have a look at my new package lasR https://github.com/r-lidar/lasR where the access to the Triangulation and easier https://r-lidar.github.io/lasR/articles/tutorial.html#triangulate

**Reply by jp136769** [reply]
Hello Jean-Romain,

Thanks for your answer! I didn't know about lasR, that's great!
